modules/standard/Math.chpl

```diff
@@ -98,13 +98,24 @@ module Math {
   pragma "fn synchronization free"
   private extern proc fabs(x: real(64)): real(64);
 
-  private proc _logBasePow2Help(in val, baseLog2) {
-    var result = -1;
-    while (val != 0) {
-      val >>= baseLog2;
-      result += 1;
+  private inline proc _logBasePow2Help(in val, baseLog2) {
+    // These are used here to avoid including BitOps by default.
+    extern proc chpl_bitops_clz_32(x: c_uint) : uint(32);
+    extern proc chpl_bitops_clz_64(x: c_ulonglong) : uint(64);
+
+    var lg2 = 0;
+
+    if numBits(val.type) <= 32 {
+      var tmp:uint(32) = val:uint(32);
+      lg2 = 32 - 1 - chpl_bitops_clz_32(tmp):int;
+    } else if numBits(val.type) == 64 {
+      var tmp:uint(64) = val:uint(64);
+      lg2 = 64 - 1 - chpl_bitops_clz_64(tmp):int;
+    } else {
+      compilerError("Case not handled in logBasePow2 for integers");
```

Comment:
What would this case be? Any way we can provide a more specific error message?

Reply:
I just left it in there in case somebody adds `int(128)` as a way to do better than just give wrong answers.

Reply:
(It's not really intended to be a user-facing error).

Reply:
How about `Integer width not handled in ...` to be more specific about what's wrong with the user's program?